src/reader.rs

```diff
@@ -350,7 +352,7 @@ impl<'a> Reader<'a> {
         header: &Header,
     ) -> Result<&'a [ReservedMemEntry]> {
         let entry_size = size_of::<ReservedMemEntry>();
-        if header.reserved_mem_offset + entry_size as u32 > header.struct_offset
+        if header.struct_offset.checked_sub(entry_size as u32) < Some(header.reserved_mem_offset)
```

Comment:
Is that just for clarity?

Reply:
No. `struct_offset` is later used to calculate the length of the slice.

`header.struct_offset - header.reserved_mem_offset` may silently overflow in release when. Note that the previous check is not strong to assert the relation of the two alone, as the addition may have itself overflown.

Reply:
That's also what the faulty file exploits. It has `reserved_mem_offset = 0xFFFF_FFF0` which is aligned but overflows to `0` in this comparison.